cdist/conf/type/__apt_mark/gencode-remote

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+#
+# 2016 Ander Punnar (cdist at kvlt.ee)
+#
+# This file is part of cdist.
+#
+# cdist is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cdist is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with cdist. If not, see <http://www.gnu.org/licenses/>.
+#
+
+if [ -f "$__object/parameter/name" ]; then
+    name="$(cat "$__object/parameter/name")"
+else
+    name="$__object_id"
+fi
+
+state="$(cat "$__object/parameter/state")"
+
+case "$state" in
+    auto|manual|hold|unhold)
+        echo "apt-mark $state $name"
```

Comment:
Is it actually possible to retrieve the mark before with an explorer?

The notion of cdist is *not* to generate code, if not necessary - so we can eventually check which changes were made by a run.